src/engine/plugins/tile.cpp

```diff
@@ -246,207 +407,326 @@ Status TilePlugin::HandleRequest(const api::TileParameters &parameters, std::str
-                {
-                    std::int32_t start_x = 0;
-                    std::int32_t start_y = 0;
+                    std::vector<NodeID> forward_node_vector;
+                    facade.GetUncompressedGeometry(edge.forward_packed_geometry_id,
+                                                   forward_node_vector);
 
-                    // Calculate the speed for this line
-                    std::uint32_t speed_kmh =
-                        static_cast<std::uint32_t>(round(length / reverse_weight * 10 * 3.6));
+                    // Skip writing for non-intersection segments
+                    if (edge.fwd_segment_position != forward_node_vector.size() - 1)
+                    {
+                        continue;
+                    }
 
-                    auto tile_line = coordinatesToTileLine(b, a, tile_bbox);
-                    if (!tile_line.empty())
+                    const auto encode_tile_point =
+                        [&point_layer_writer, &edge, &id](const detail::FixedPoint &tile_point,
+                                                          const detail::TurnData &point_turn_data,
+                                                          std::int32_t &start_x,
+                                                          std::int32_t &start_y)
+                    {
+                        protozero::pbf_writer feature_writer(point_layer_writer,
+                                                             util::vector_tile::FEATURE_TAG);
+                        // Field 3 is the "geometry type" field.  Value 1 is "point"
+                        feature_writer.add_enum(
+                            util::vector_tile::GEOMETRY_TAG,
+                            util::vector_tile::GEOMETRY_TYPE_POINT); // geometry type
+                        // Field 1 for the feature is the "id" field.
+                        feature_writer.add_uint64(util::vector_tile::ID_TAG, id++); // id
+                        {
+                            // See above for explanation
+                            protozero::packed_field_uint32 field(
+                                feature_writer, util::vector_tile::FEATURE_ATTRIBUTES_TAG);
+
+                            field.add_element(0); // "bearing_in" tag key offset
+                            field.add_element(point_turn_data.in_angle_offset);
+                            field.add_element(1); // "bearing_out" tag key offset
+                            field.add_element(point_turn_data.out_angle_offset);
+                            field.add_element(2); // "weight" tag key offset
+                            field.add_element(point_turn_data.weight_offset);
+                        }
+                        {
+                            protozero::packed_field_uint32 geometry(
+                                feature_writer, util::vector_tile::FEATURE_GEOMETRIES_TAG);
+                            encodePoint(tile_point, geometry, start_x, start_y);
```

Comment:
@danpat cool — I sort of suspected as much but figured I could just start with the copypasta and then left it in there, will remove 👍